core/src/main/java/org/elasticsearch/index/translog/TranslogReader.java

```diff
@@ -19,45 +19,30 @@
 
 package org.elasticsearch.index.translog;
 
-import org.apache.lucene.codecs.CodecUtil;
-import org.apache.lucene.index.CorruptIndexException;
-import org.apache.lucene.index.IndexFormatTooNewException;
-import org.apache.lucene.index.IndexFormatTooOldException;
-import org.apache.lucene.store.AlreadyClosedException;
-import org.apache.lucene.store.InputStreamDataInput;
-import org.apache.lucene.util.BytesRef;
-import org.apache.lucene.util.RamUsageEstimator;
 import org.elasticsearch.ElasticsearchException;
 import org.elasticsearch.common.io.stream.ByteBufferStreamInput;
-import org.elasticsearch.common.io.stream.InputStreamStreamInput;
 
-import java.io.Closeable;
 import java.io.IOException;
 import java.nio.ByteBuffer;
-import java.nio.channels.Channels;
 import java.nio.channels.FileChannel;
-import java.nio.file.Files;
 import java.nio.file.Path;
-import java.util.concurrent.atomic.AtomicBoolean;
 
 /**
  * A base class for all classes that allows reading ops from translog files
  */
-public abstract class TranslogReader implements Closeable, Comparable<TranslogReader> {
-    public static final int UNKNOWN_OP_COUNT = -1;
-    private static final byte LUCENE_CODEC_HEADER_BYTE = 0x3f;
-    private static final byte UNVERSIONED_TRANSLOG_HEADER_BYTE = 0x00;
+public abstract class TranslogReader implements Comparable<TranslogReader> {
```

Comment:
Can we rename this class to `BaseTranslogReader` and rename `ImmutableTranslogReader` to `TranslogReader`?